### PR TITLE
Use public getter for IDGraphHash

### PR DESCRIPTION
**Pull request by Kailai-Wang**
### Context

See the linked issue.

This PR:
- dismisses the on-parachain IDGraph storage in favor of a public getter
- adjusts ts-test
- updates some crates to supersede dependabot PRs

It's a breaking change for the client.

### CODE REVIEW

File: pallets/identity-management/src/lib.rs [hunk 1]
```diff
@@ -96,22 +94,18 @@ pub mod pallet {
 		//       Can we retrieve that extrinsic hash in F/E?
 		IdentityLinked {
 			account: T::AccountId,
-			id_graph_hash: H256,
 			req_ext_hash: H256,
```
Comment (jonalvarezz) [inline pallets/identity-management/src/lib.rs]: Would calling the public getter be cheaper than including the hash in the event's response? I can imagine clients calling the public id_graph_hash getter right after each IdentityManagement operation.

File: tee-worker/litentry/pallets/identity-management/src/lib.rs [hunk 2]
```diff
@@ -295,7 +295,10 @@ pub mod pallet {
 		}
 
 		// get the whole IDGraph, sorted by `link_block` (earliest -> latest)
-		pub fn get_id_graph(who: &Identity) -> IDGraph<T> {
+		//
+		// TODO: shall we change the return type to Option<IDGraph<T>> and return
```
Comment (jonalvarezz) [inline tee-worker/litentry/pallets/identity-management/src/lib.rs:299]: sounds good

File: tee-worker/litentry/pallets/identity-management/src/lib.rs [hunk 3]
```diff
@@ -304,10 +307,14 @@ pub mod pallet {
...
 		}
 
-		pub fn all_id_graph_hash() -> Vec<(Identity, H256)> {
-			IDGraphLens::<T>::iter_keys()
-				.map(|k| (k.clone(), H256::from(blake2_256(&Self::get_id_graph(&k).encode()))))
-				.collect()
+		// get the IDGraph hash of the given `who`
+		pub fn id_graph_hash(who: &Identity) -> Option<H256> {
+			let id_graph = Self::id_graph(who);
+			if id_graph.is_empty() {
```
Comment (jonalvarezz) [inline tee-worker/litentry/pallets/identity-management/src/lib.rs:313]: if `who` is not a prime identity but a leaf of the id_graph, would the result be empty too?

File: tee-worker/ts-tests/integration-tests/common/di-utils.ts [hunk 4]
```diff
@@ -257,34 +256,50 @@ export async function createSignedTrustedCallActivateIdentity(
...
+    primeIdentity: LitentryPrimitivesIdentity
+): Promise<H256> => {
+    const getterPublic = createPublicGetter(
+        context.api,
+        ['id_graph_hash', '(LitentryIdentity)'],
+        primeIdentity.toHuman()
+    );
+    const getter = context.api.createType('Getter', { public: getterPublic }) as unknown as Getter; // @fixme 1878
+    const res = await sendRequestFromGetter(context, teeShieldingKey, getter);
+    const hash = context.api.createType('Option<Bytes>', hexToU8a(res.value.toHex())).unwrap();
```
Comment (jonalvarezz) [inline tee-worker/ts-tests/integration-tests/common/di-utils.ts:301]: nitpicking: it should work without hex -> u8a 
```suggestion
    const hash = context.api.createType('Option<Bytes>', res.value.toHex()).unwrap();
```

**Review by jonalvarezz — APPROVED** [review]
amazing! 

overall looking good to time, although not certain about removing `id_graph_hash` from event's reponses

File: tee-worker/litentry/pallets/identity-management/src/lib.rs (hunk 3 above)
Comment (Kailai-Wang) [reply tee-worker/litentry/pallets/identity-management/src/lib.rs:313]: > if `who` is not a prime identity but a leaf of the id_graph, would the result be empty too?

Yes, will be `None`

File: pallets/identity-management/src/lib.rs (hunk 1 above)
Comment (Kailai-Wang) [reply pallets/identity-management/src/lib.rs]: In short - I don't think client needs to listen to the event, the synchronous DI response contains everything that a client needs.

However, if you prefer having it in parachain event for tracking purpose, I'm fine too

**Review by arieljbirnbaum — COMMENTED** [review]
No objections here; awesome improvement <3

File: tee-worker/ts-tests/integration-tests/common/di-utils.ts (hunk 4 above)
Comment (Kailai-Wang) [reply tee-worker/ts-tests/integration-tests/common/di-utils.ts:301]: No it doesn't, I believe I had similar questions 😂 

without `hexToU8a` I got:
```
Error: createType(LinkIdentityResult):: decodeU8aStruct: failed at 0x040110000000000000… on mutated_id_graph (index 1/2): {"ciphertext":"Bytes","aad":"Bytes","nonce":"[u8;12]"}:: decodeU8aStruct: failed at 0x0000… on nonce (index 3/3): [u8;12]:: Expected input with 12 bytes (96 bits), found 2 bytes
```

File: pallets/identity-management/src/lib.rs (hunk 1 above)
Comment (Kailai-Wang) [reply pallets/identity-management/src/lib.rs]: >  I can imagine clients calling the public id_graph_hash getter right after each IdentityManagement operation.

The client doesn't have to. It's included in each DI request that mutates the IDGraph, see `trusted_call_result.rs`

**Review by BillyWooo — APPROVED** [review]

File: pallets/identity-management/src/lib.rs (hunk 1 above)
Comment (Kailai-Wang) [reply pallets/identity-management/src/lib.rs]: After a second thought I reckon it's beneficial to include these hashes for tracking purposes.

File: tee-worker/ts-tests/integration-tests/common/di-utils.ts (hunk 4 above)
Comment (jonalvarezz) [reply tee-worker/ts-tests/integration-tests/common/di-utils.ts:301]: :flip-table:

File: pallets/identity-management/src/lib.rs (hunk 1 above)
Comment (jonalvarezz) [reply pallets/identity-management/src/lib.rs]: Thank, Kai!

**Pull request merged**